code/processes/dqc.q

```diff
@@ -0,0 +1,290 @@
+    procsdown:(h`procname) where 0N = h`w;
+    if[count procsdown;.dqe.updresultstab[runtype;idnum;0Np;0b;"error:process is down or has lost its handle";`failed;params]'[procsdown]];
+  ];
+  if[params`comp;
+    /- fail if comparison process is in list of processes to check against
+    if[(params`compproc) in h`procname;
+      .lg.e[`runcheck;"Can't compare process with itself"];
+      .dqe.updresultstab[runtype;idnum;0Np;0b;"error:compare process can't be compared with itself";`failed;params]'[h`procname];
+      :()];
+
+    params,:(enlist `compresproc)!enlist `$"," sv string h`procname;
+    /- obtain handle for comparison process
+    comph:.dqe.gethandles[params`compproc];
+    h:h,'comph;
+
+    proccount:count h`procname;
+    params,:(enlist `compcount)!enlist proccount;
+
+    .lg.o[`runcheck;(string params`compcount)," process will be checked for this comparison"];
+    .dqe.initstatusupd[runtype;idnum;fn;params;(`$"," sv string  r[;0]),params`compresproc];
+
+    if[any[null h`w]|any null r[;1];
+      .lg.e[`runcheck;"unable to compare as process down or missing handle"];
+      .dqe.updresultstab[runtype;idnum;0Np;0b;"error:unable to compare as process down or missing handle";`failed;params;params`compresproc];
+      :()];
+   ]
+  /- check if any handles exist, if not exit function
+  if[0=count h;.lg.e[`runcheck;"cannot open handle to any given processes"];:()];
+  .dqe.getresult[runtype;value fn;(),params;idnum]'[h[`procname];h[`w]]
+  }
+
+results:([]id:`long$();funct:`$();params:`$();procs:`$();procschk:`$();starttime:`timestamp$();endtime:`timestamp$();result:`boolean$();descp:();chkstatus:`$();chkruntype:`$());
+
+loadtimer:{[DICT]
+  .lg.o[`dqc;("Loading check - ",(string DICT[`action])," from configtable into timer table")];
+  /- Accounting for potential multiple parameters
+  DICT[`params]: value DICT[`params];
+  DICT[`proc]: value DICT[`proc];
+  /- function that will be used in timer
+  functiontorun:(`.dqe.runcheck;`scheduled;DICT`checkid;.Q.dd[`.dqc;DICT`action];DICT`params;DICT`proc);
+  /- Determine whether the check should be repeated
+  $[DICT[`mode]=`repeat;
+    .timer.repeat[DICT`starttime;DICT`endtime;DICT`period;functiontorun;"Running check on ",string DICT`proc];
+    .timer.once[DICT`starttime;functiontorun;"Running check once on ",string DICT`proc]]
+  }
+
+/- rerun a check manually
+reruncheck:{[chkid]
+  d:exec action, params, proc from .dqe.configtable where checkid=chkid;
+  .lg.o[`dqc;("re-running check ",string(d[`action])," manually")];
+  d[`params]: value d[`params][0];
+  d[`proc]: value raze d[`proc];
+  /- input man argument is `manual or `scheduled indicating manul run is on or off
+  .dqe.runcheck[`manual;chkid;.Q.dd[`.dqc;d`action];d`params;d`proc];
+  }
+
+\d .
+
+.dqe.currentpartition:.dqe.getpartition[];
+
+.servers.CONNECTIONS:`ALL                                                                                       /- set to nothing so that is only connects to discovery
```

Comment:
:eyes: Also as Jonny said this comment needs looked at, doesn't seem to be accurate.

Reply:
this one i figured was a global variable, so does that mean that the comments need to be
```
VARIABLE: 42                                 // comment
```
?

Reply:
In that case this should be moved to the top of the file.

Reply:
ok, cool.

Also, just asked daniel and the code is correct, it's supposed to connect to everything, not just disocvery